settings.php

```diff
@@ -76,10 +76,14 @@
 } else {
     $piHoleInterface = "unknown";
 }
-if (isset($setupVars["IPV4_ADDRESS"])) {
-    $piHoleIPv4 = $setupVars["IPV4_ADDRESS"];
-} else {
-    $piHoleIPv4 = "unknown";
+
+// get the gateway address for the default route on the Pi-hole interface
+if ($piHoleInterface !== "unknown") {
+    $IPv4GW=shell_exec("ip -4 route | grep default | grep '${piHoleInterface}' | cut -d ' ' -f 3");
+}
+// if the route did not return anything or $piHoleInterface is unknown
+if (empty($IPv4GW)) {
+    $IPv4GW = "unknown";
 }
```

Comment:
I wouldn't do any interface filtering here, we should just use the `default` gateway in all cases.
After all, we use this address just as a hint. Using the default network gateway's address will be the right thing in virtually all cases.

Also, because the default gateway is the route to use as the last resort if none of the other routes match. So by that definition, there can be only one **active** default route at a time so grepping for `default` and then again for an interface may result in no result (even when there could be one).

```suggestion
// get the gateway address for the default route on the Pi-hole interface
$IPv4GW = shell_exec("ip -4 route | grep default | cut -d ' ' -f 3");

// if the route did not return anything (shouldn't happen)
if (empty($IPv4GW)) {
    $IPv4GW = "unknown";
}
```

Reply:
Thinking about that we'll need to port this over to FTL anyway at some point (when getting rid of PHP), we could also use what is proposed in FTL's new branch `new/default_gateway`

``` plain
$ echo ">gateway" | ncat 127.0.0.1 4711
192.168.0.1 enp2s0
---EOM---
```

instead of using `shell_exec()` here to run three commands connected by a pipe in a shell.

Reply:
We also could use `ip -4 route show default |  cut -d ' ' -f 3`.

Reply:
I think I'll go with the new API endpoint. I'll see if I can test/review the FTL PR tonight and change this one accordingly.

Reply:
Using FTL' s new API now.